Issue142 alternative container diagrams

mortenwh opened the pull request:
#142. Needs discussion/review. Closes #144, #146, #147.

mortenwh commented:
@mpejcoch - it seems the new diagram under "Second iteration with more containers and functionality" does not compile now. Could it be some paths that are set incorrectly?

mpejcoch commented:
There are some issues yes, the architecture also doesn't work from the link on the main page, but is available when explicitly going to: `https://s-enda-documentation.readthedocs.io/en/issue142_alternative_container_diagrams/architecture/architecture.html`

mortenwh commented:
ok - will you handle that?

On Thu, May 7, 2020 at 2:53 PM Martin Grønlien Pejcoch <
notifications@github.com> wrote:

> There are some issues yes, the architecture also doesn't work from the
> link on the main page, but is available when explicitly going to:
> https://s-enda-documentation.readthedocs.io/en/issue142_alternative_container_diagrams/architecture/architecture.html
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/metno/S-ENDA-documentation/pull/160#issuecomment-625236664>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA2UBLRWTNZP4TFLVV2AS3RQKVLVANCNFSM4MVWHNRA>
> .
>


-- 
Morten Wergeland Hansen, PhD
Meteorologisk Institutt / Norwegian Meteorological Institute
T.: (+47) 915 47 844

mpejcoch commented:
> ok - will you handle that?

That is the plan. Worst case I just revert it to the flat structure again. If you have any ideas on what's wrong let me know. I'm quite new to Sphinx. What I did is that I've moved the .rst and .puml files and changed:
```
.. toctree::
   :maxdepth: 2
   :caption: Contents

   architecture/architecture
   metadata_management_heritage
   use_case_descriptions/use_cases
   catalog_service_API
   dm_recipes
```
to point to the subdirectories, but it doesn't work as expected I am afraid.

mortenwh commented:
Ok - yes, I noticed that. Not sure how to do it, and I would prefer to
focus on some tests now. Hope that's ok :)

On Thu, May 7, 2020 at 3:07 PM Martin Grønlien Pejcoch <
notifications@github.com> wrote:

> ok - will you handle that?
>
> That is the plan. Worst case I just revert it to the flat structure again.
> If you have any ideas on what's wrong let me know. I'm quite new to Sphinx.
> What I did is that I've moved the .rst and .puml files and changed:
>
> .. toctree::
>    :maxdepth: 2
>    :caption: Contents
>
>    architecture/architecture
>    metadata_management_heritage
>    use_case_descriptions/use_cases
>    catalog_service_API
>    dm_recipes
>
> to point to the subdirectories, but it doesn't work as expected I am
> afraid.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/metno/S-ENDA-documentation/pull/160#issuecomment-625243858>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA2UBL2T7M6AHSLFMICRN3RQKW73ANCNFSM4MVWHNRA>
> .
>


-- 
Morten Wergeland Hansen, PhD
Meteorologisk Institutt / Norwegian Meteorological Institute
T.: (+47) 915 47 844

mortenwh commented on source/search-context.puml:
```diff
@@ -0,0 +1,32 @@
+    '  Person(user, "General user")
+   '}
+
+    System(msys, "my system")
```
what is this? @havardf @mpejcoch

mpejcoch replied:
I think that was an attempt to get rid of the tripple "=" over that arrow which I added unintentionally. I fix.

mortenwh replied:
why do those "=" appear? I got it also now...

mortenwh commented on source/search-context.puml:
```diff
@@ -0,0 +1,32 @@
+      System(sendafind_nodes, "S-ENDA Metadata Service Nodes")
+   }
+
+   Rel(users, msys, "Users search portals", "Web-UI/API")
```
this relation doesn't make sense to me.. is it an error? I thought the relation should be to the Portals boundary(?)

mortenwh replied:
@havardf @mpejcoch

mortenwh commented on source/container.puml:
```diff
@@ -22,6 +24,10 @@ Rel(geoasset_api, eventengine, "Listens in order to get last updated info", "Clo
 }
 
 Rel(geonorge, csapi, "Harvests metadata", "CSW")
+Rel(datanorge, api_met_no, "Harvest api.met.no metadata catalog", "HTTP (DCAT)")
```
Shouldn't there also be a connection to the Dynamic Geo-Assets API here, for registration? @havardf 

Apart from that, maybe we can remove this diagram now? Shouldn't it anyway be exchanged with a combination of second and third iterations?

havardf replied:
Well, we agreed that registrering a catalog is such a rare event that it seems like overkille to have an api for it. A config file thats manually updated is enough. And manual updates doesnt really fit into this diagram I think.

I guess it would. Its anyway better to not have multiple similar diagrams, because its confusing who is authoritative.

mortenwh commented on source/newer_S_ENDA_metadata_service_container.puml:
```diff
@@ -0,0 +1,56 @@
+Container(updater, "Metadata Updater", "", "Provides functionality to compare provided and existing content, delete content, and create and update content in the discovery and configuration metadata store.")
+
+Container(usage_stats, "Web Page with Usage Statistics", "", "Provides functionality to display dataset usage statistics on a web page.")
+Container(prod_status, "Web Page with Production status", "", "Provides functionality to display dataset production status on a web page.")
```
Doesn't this rather belong in the Production Hubs system? @mpejcoch
